libs/VideoKit/source/VideoKit.cpp

```diff
@@ -4,34 +4,81 @@
 
 #include "VideoKit.h"
 
+#include "rtos/ThisThread.h"
+
+#include "FileManagerKit.h"
+
 using namespace leka;
+using namespace std::chrono_literals;
 
 void VideoKit::initializeScreen()
 {
-	// nothing to do
-}
+	_video.initialize();
 
-void VideoKit::turnOn()
-{
-	// nothing to do
-}
+	_video.setBrightness(1.F);
+	_video.clearScreen();
 
-void VideoKit::turnOff()
-{
-	// nothing to do
+	_thread.start(mbed::Callback(this, &VideoKit::run));
 }
 
 void VideoKit::displayImage(const std::filesystem::path &path)
 {
-	// nothing to do
+	if (auto file = FileManagerKit::File {path}; file.is_open()) {
+		_event_flags.set(flags::STOP_VIDEO_FLAG);
+
+		rtos::ThisThread::sleep_for(100ms);
+
+		_video.displayImage(file);
+		_lcd.turnOn();
+
+		file.close();
+	}
 }
 
 void VideoKit::playVideo(const std::filesystem::path &path, bool must_loop)
 {
-	// nothing to do
+	if (auto file = FileManagerKit::File {path}; file.is_open()) {
+		file.close();
+
+		_event_flags.set(flags::STOP_VIDEO_FLAG);
+
+		_current_path = path;
+		_must_loop	  = must_loop;
+
+		rtos::ThisThread::sleep_for(100ms);
+		_event_flags.set(flags::START_VIDEO_FLAG);
+
+		_lcd.turnOn();
+	}
 }
 
 void VideoKit::stopVideo()
 {
-	// nothing to do
+	_event_flags.set(flags::STOP_VIDEO_FLAG);
+}
+
+void VideoKit::run()
+{
+	auto file = FileManagerKit::File {};
+
+	auto has_to_stop_video = [this] { return (_event_flags.get() & flags::STOP_VIDEO_FLAG) == flags::STOP_VIDEO_FLAG; };
```

Comment:
"must" et "have to" veulent globalement dire la même chose, cependant en programmation on préfèrera "must", see https://semver.org/#semantic-versioning-specification-semver ou https://datatracker.ietf.org/doc/html/rfc2119


```suggestion
	auto must_stop_video = [this] { return (_event_flags.get() & flags::STOP_VIDEO_FLAG) == flags::STOP_VIDEO_FLAG; };
```